### ISSUE TITLE
[css-pseudo] more clearly define which pseudo-elements are tree-abiding or part-like

### ISSUE THREAD

**Issue by dbaron** [w3c/csswg-drafts#10794]
css-pseudo-4 [defines](https://drafts.csswg.org/css-pseudo-4/#treelike) the terms "tree-abiding pseudo-element" and  (a stricter subset) "part-like pseudo-element".  Some pseudo elements fall into these categories.

These cause various things to be allowed.  In particular, I think the current list is that tree-abiding pseudo-elements are:

* allowed to change the results of `getComputedStyle`
* allowed after `::slotted()`
* support the `content` property

and part-like pseudo-elements do all of the above and also:

* support any CSS properties except those that say otherwise
* can be followed by most pseudo-classes and any pseudo-elements (not yet specified, see #10083)

However, there are multiple specifications that define pseudo-elements, some of which define pseudo-elements that probably belong in one of these categories but don't say so, and some of which define pseudo-elements as tree-abiding when they may want the newer part-like definition instead.

We should audit all pseudo-elements and make sure that they're defined appropriately.  We should probably also get a group resolution on the result of that audit, since the changes in feature support for many of these pseudo-elements are substantive.  (The initial introduction of the part-like term defined a bunch of pseudo-elements as part-like, and I think we could probably use a resolution to approve that.)

**Comment by dbaron** [comment]
I think we also need to consider whether more information is needed in the spec to define a part-like pseudo-element (such as what type of element it is, which can affect which pseudo classes it supports).

**Comment by tabatkins** [comment]
>  I think we also need to consider whether more information is needed in the spec to define a part-like pseudo-element (such as what type of element it is, which can affect which pseudo classes it supports).

Yeah, this information is needed. I think we can *default* to making them a "generic" element which possesses no special qualities, but we should introduce a term making it easy to have a part-like "defer" to a real element (possibly in another tree, including a fully UA-generated/controlled one) for the purpose of its properties. That latter would cover ::part(), obviously, and also things like the new base-appearance form pseudos.

**Comment by dbaron** [comment]
I think I'm inclined to suggest the following (though there are a bunch that I'm unsure about):

part-like:
* `::part()`
* `::file-selector-button`
* `::details-content`
* `::picker()`
* `::slotted()` (?)

tree-abiding:
* `::after` and `::before`
* `::backdrop` (↓?)
* `::marker`
* `::placeholder`
* `::scroll-marker-group` and `::scroll-marker` (?)
* `::view-transition`, `::view-transition-group()`, `::view-transition-image-pair()`, `::view-transition-new()`, `::view-transition-old()` (↑?)

neither:
* `::attr()`
* `::content`
* `::cue`, `::cue()`, `::cue-region`, `::cue-region()` (?)
* `::first-letter`
* `::first-line`
* `::selection`, `::grammar-error`, `::spelling-error`, `::highlight()`
* `::nth-fragment()`
* `::prefix`, `::postfix`
* `::shadow` (?)
* `::target-text`
